amaranth/vendor/_lattice_nexus.py

```diff
@@ -0,0 +1,884 @@
+
+from abc import abstractproperty
+
+from ..hdl import *
+from ..build import *
+
+__all__ = ["LatticeNexusPlatform"]
+
+class LatticeNexusPlatform(TemplatedPlatform):
+    """
+    .. rubric:: Oxide toolchain
+
+    Required tools:
+        * ``yosys``
+        * ``nextpnr-nexus``
+        * ``prjoxide``
+
+    FIXME The environment is populated by running the script specified in the environment variable
+    ``AMARANTH_ENV_OXIDE``, if present.
+
+    Available overrides:
+        * ``verbose``: enables logging of informational messages to standard error.
+        * ``read_verilog_opts``: adds options for ``read_verilog`` Yosys command.
+        * ``synth_opts``: adds options for ``synth_nexus`` Yosys command.
+        * ``script_after_read``: inserts commands after ``read_ilang`` in Yosys script.
+        * ``script_after_synth``: inserts commands after ``synth_nexus`` in Yosys script.
+        * ``yosys_opts``: adds extra options for ``yosys``.
+        * ``nextpnr_opts``: adds extra options for ``nextpnr-nexus``.
+        * ``prjoxide_opts``: adds extra options for ``prjoxide``.
+        * ``add_preferences``: inserts commands at the end of the LPF file.
+
+    Build products:
+        * ``{{name}}.rpt``: Yosys log.
+        * ``{{name}}.json``: synthesized RTL.
+        * ``{{name}}.tim``: nextpnr log.
+        * ``{{name}}.config``: ASCII bitstream.
+        * ``{{name}}.bit``: binary bitstream.
+        * ``{{name}}.xcf``: JTAG programming vector.
+
+    .. rubric:: Radiant toolchain
+
+    Required tools:
+        * ``yosys`` # optional
+        * ``radiantc``
+        * ``programmer`` # optional
```

Comment:
I'm not sure how this would actually work? A required optional tool?

Reply:
removed --now mirrors what @richardeoin had (and what's actually required).